source/fuzz/fuzzer_pass_add_parameters.cpp

```diff
@@ -70,14 +72,76 @@ void FuzzerPassAddParameters::Apply() {
-          FindOrCreateZeroConstant(
-              type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)],
-              true),
-          GetFuzzerContext()->GetFreshId()));
+      auto current_type_id =
+          type_candidates[GetFuzzerContext()->RandomIndex(type_candidates)];
+      auto current_type =
+          GetIRContext()->get_type_mgr()->GetType(current_type_id);
+
+      // Consider pointer types separately.
+      if (current_type->kind() == opt::analysis::Type::kPointer) {
+        auto storage_class = fuzzerutil::GetStorageClassFromPointerType(
+            GetIRContext(), current_type_id);
+        uint32_t pointee_type_id =
+            (GetIRContext()->get_def_use_mgr()->GetDef(current_type_id))
+                ->GetSingleWordInOperand(1);
+        uint32_t variable_id = 0;
+
+        // Make the transformation applicable.
+        switch (storage_class) {
+          case SpvStorageClassFunction: {
+            for (auto* instr :
+                 fuzzerutil::GetCallers(GetIRContext(), function.result_id())) {
+              auto block = GetIRContext()->get_instr_block(instr);
+              auto function_id = block->GetParent()->result_id();
+              // If there is no available local variable, then create one. The
+              // available local variable must be marked with the fact
+              // PointeeValueIsIrrelevant.
+              variable_id = fuzzerutil::MaybeGetLocalVariable(
+                  GetIRContext(), current_type_id, function_id);
+              if (!variable_id ||
+                  !GetTransformationContext()
+                       ->GetFactManager()
+                       ->PointeeValueIsIrrelevant(variable_id)) {
+                ApplyTransformation(TransformationAddLocalVariable(
+                    GetFuzzerContext()->GetFreshId(), current_type_id,
+                    function_id,
+                    FindOrCreateZeroConstant(pointee_type_id, true), true));
+              }
+            }
+          } break;
+          case SpvStorageClassPrivate:
+          case SpvStorageClassWorkgroup: {
+            // If there is no available global variable, then create one. The
+            // available global variable must be marked with the fact
+            // PointeeValueIsIrrelevant.
+            variable_id = fuzzerutil::MaybeGetGlobalVariable(GetIRContext(),
+                                                             current_type_id);
+            if (!variable_id || !GetTransformationContext()
+                                     ->GetFactManager()
+                                     ->PointeeValueIsIrrelevant(variable_id)) {
+              ApplyTransformation(TransformationAddGlobalVariable(
+                  GetFuzzerContext()->GetFreshId(), current_type_id,
+                  storage_class,
+                  FindOrCreateZeroConstant(pointee_type_id, true), true));
+            }
+          } break;
+          default:
+            break;
```

Comment:
It might be better to add an assert here saying that storage class is not supported. This will make it easier to identify all the places we need to adjust if we decide to add support for more storage classes.